### ISSUE TITLE
wifi-connect --activity-timeout option does not seem to work

### ISSUE THREAD

**Issue by bbinet** [resin-io/resin-wifi-connect#186]
I'm trying to integrate resin-wifi-connect in my project, and I would like to use the `--activity-timeout` option (see: https://github.com/resin-io/resin-wifi-connect/blob/master/docs/command-line-arguments.md#options), but the timeout never expires in my case.

Here is my start.sh script:
```
#!/usr/bin/env bash
export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket
while true
do
    iwgetid -r
    if [ $? -ne 0 ]; then
        printf 'Starting WiFi Connect\n'
        ./wifi-connect --portal-ssid myssid --portal-passphrase mypass --activity-timeout 600
    fi
    sleep 10
done
```
I want `wifi-connect` to restart every 10 minutes for 2 reasons:
- to let resinos network-manager the opportunity to connect to the already existing wifi configuration (in case `wifi-connect` has started because the wifi was temporarily not available).
- and so that `wifi-connect` can scan again the available wifis around and discover potential new wifis.

But in my experiment, the `wifi-connect` program never ends.
If I `pkill wifi-connect`, the loop in my `start.sh` script continues and resinos network-manager is able to connect to its already existing wifi configuration.

So I'm wondering if the `--activity-timeout` option is actually implemented, or maybe I misunderstand the actual meaning of `--activity-timeout`?

**Comment by majorz** [comment]
I checked now on RPi3 with a minimal start script and minimal Dockerfile (the ones the repo) with v4.0.6 and `./wifi-connect --activity-timeout 10` and it exited after 10 seconds:
```
Access point 'WiFi Connect' created
Starting HTTP server on 192.168.42.1:80
Timeout reached. Exiting...
Stopping access point 'WiFi Connect'...
Access point 'WiFi Connect' stopped
```

Your start script uses the `--activity-timeout` option correctly, so let's see what differences we may have in the running setup to narrow down the cause. Can you please first run the application from a terminal session directly (without a start script) with `./wifi-connect --activity-timeout 10` so that we narrow it down more (do not forget to do `export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket` if you are running it in a resinOS container).

**Comment by majorz** [comment]
BTW I need to add some logging for the `--activity-timeout` option, but solve this one first.

**Comment by bbinet** [comment]
@majorz thanks for your quick reply.
It is not easy for me to run `wifi-connect` from a terminal session directly, because it runs on a rpi zero w (without ethernet, so I will loose the connection when starting `wifi-connect`).
But I can try to build an image and test from a rpi3, but if it work it won't explain why it does not work on the zero w...

**Comment by bbinet** [comment]
@majorz if you add some logs to `wifi-connect` first, it could actually help the debugging on headless systems

BTW, how wifi-connect will behave if someone connects to captive portal within the 10min interval, then do nothing, and leave.
Will wifi-connect wait again for the next 10min timeout or will it wait forever because someone was connected once? (this actually what I did initially: access to the captive portal, then leave without changing the wifi configuration)

**Comment by majorz** [comment]
That explains it. It will wait forever once you have connected to the captive portal. I am not sure whether we need to change its behaviour though and how exactly. Do you have any ideas in this regard?

**Comment by bbinet** [comment]
It does not need to be the same activity timeout, but it will be useful to be able to configure wifi-connect with a timeout to be sure it will not run indefinitely.
But I think it might makes sense to keep the same activity timeout: wifi-connect could just reset the timeout to its original value when it detects some activity on the captive portal, so if the user leaves the portal without changing anything, the timeout will finally expires 10min later (if activity timeout was set to 600).

**Comment by majorz** [comment]
I agree, it makes more sense to reset the activity flag that way. I will have to consult on this with the rest of the team, so that we do not miss something, and I will ping you once we discuss it.

**Comment by bbinet** [comment]
ok, thanks.

**Comment by majorz** [comment]
I raised the topic for our company calls. I should have an answer somewhere next week, since this one is a focus/no-calls week.

**Comment by bbinet** [comment]
thanks again :)

**Comment by majorz** [comment]
@bbinet just to update you on this one, as we just discussed it on a call. We will change the behavior in the way that you suggested, so let's keep the issue open.

**Comment by bbinet** [comment]
@majorz That's great, this will be very helpful.

**Comment by bbinet** [comment]
@majorz I was wondering how this use case would work when controlling wifi-connect throught its api like https://github.com/resin-io-projects/resin-wifi-connect-api ?
My point is if nobody is interacting with the wifi-connect api, NetworkManager should regularly try to reconnect using its current wifi profiles so that the connection can be restored (for example if the access point was temporarily down). 
Maybe the api should be extended to support starting/stopping the wifi access point dynamically?

**Comment by majorz** [comment]
We will add an option for running WiFi Connect indefinitely and this fits that nicely. I will explore it once I get back to working on the API soon.
